### PR TITLE
Feature: Reusable Consul stub

### PR DESCRIPTION
**Pull request by fmitchell-r7**
This is preliminary work for #61. The Consul stub has been refactored so it's usable outside the Consul tests. This will enable the Conqueso tests to mock data from the Consul plugin and ensure it's formatted correctly. The sub has also been documented at the API level.

### CODE REVIEW

File: test/utils/consul-stub.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
+'use strict';
```
Comment (rvanburen-r7) [inline test/utils/consul-stub.js]: Quick question, shouldn't we have `/* eslint-env mocha */

Comment (fmitchell-r7) [inline test/utils/consul-stub.js]: I don't think it's necessary, no. As far as I know, mocha won't run up anything in the "./utils" folder by default, and this file shouldn't be using anything mocha specific.

Comment (rvanburen-r7) [inline test/utils/consul-stub.js]: Ah, thanks for the explanation!

**Comment by rvanburen-r7** [comment]
LGTM, but you may still want another pair of :eyes: on this.

**Comment by dgreene-r7** [comment]
Very cool. Nice job!

**Pull request merged**